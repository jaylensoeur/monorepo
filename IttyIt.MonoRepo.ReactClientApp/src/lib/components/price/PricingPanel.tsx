import * as React from "react";
import useStyles from "./pricingPanelStyles";

type Props = {
    priceDescription?: string
    currency?: string
    currencySymbol: string
    price: number
    savingApproxPrice?: number
};

const PricingPanel = ({
      priceDescription,
      currency,
      price,
      currencySymbol,
      savingApproxPrice
}: Props) => {
    const classes = useStyles();
    return (
        <div className={classes.pricePanel}>
            <sup data-testid="price-description">{priceDescription} {currency ? `(${currency})` : ''}</sup>
            <div data-testid="currency-price">
                <sup className={classes.currencySymbol}>{currencySymbol}</sup>
                <span data-testid="price" className={classes.price}>{price}</span>
            </div>
        <div data-testid="approx-savings">
            {savingApproxPrice ? (
                <span className={classes.savings}>
                    Save ${savingApproxPrice}
                    <sub className={classes.savingsApproxSymbol}>~</sub>
                </span>
            ) : ''}
        </div>
    </div>)
};

export default PricingPanel;
