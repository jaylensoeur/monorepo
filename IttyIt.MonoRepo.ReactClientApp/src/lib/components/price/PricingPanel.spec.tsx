import * as React from "react";
import { create } from "react-test-renderer";
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'

import PricingPanel from "./PricingPanel";

describe("<PricingPanel />", () => {
    it("should render", async () => {
        const price = 250;
        const currencySymbol = "$";
        const component = create(<PricingPanel price={price} currencySymbol={currencySymbol} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it("should see a currency price", async () => {
        const price = 250;
        const currencySymbol = "$";
        render(<PricingPanel price={price} currencySymbol={currencySymbol} />);
        const priceSection = await screen.findByTestId('price');
        const currencyPriceSection = await screen.findByTestId('currency-price');
        const approxSavingsSection = await screen.findByTestId('approx-savings');

        expect(priceSection).toHaveTextContent(price.toString());
        expect(approxSavingsSection).toHaveTextContent('');
        expect(currencyPriceSection).toHaveTextContent(/\$250/i)

    });

    it("should see a approx savings", async () => {
        const price = 250;
        const approxSaving = 25;
        const currencySymbol = "$";
        render(<PricingPanel price={price} savingApproxPrice={approxSaving} currencySymbol={currencySymbol} />);
        const approxSavingsSection = await screen.findByTestId('approx-savings');

        expect(approxSavingsSection).toHaveTextContent(/Save \$25~/i);
    });

    it("should see a price description", async () => {
        const price = 250;
        const priceDescription = "1 night only";
        const currencySymbol = "$";
        render(<PricingPanel price={price} priceDescription={priceDescription} currencySymbol={currencySymbol} />);
        const priceDescriptionSection = await screen.findByTestId('price-description');

        expect(priceDescriptionSection).toHaveTextContent(`${priceDescription}`);
    });

    it("should see the currency used", async () => {
        const price = 250;
        const priceDescription = "1 night only";
        const currency = "USD";
        render(<PricingPanel price={price} priceDescription={priceDescription} currency={currency} currencySymbol="$" />);
        const priceDescriptionSection = await screen.findByTestId('price-description');

        expect(priceDescriptionSection).toHaveTextContent(/1 night only \(USD\)$/i);
    });
});
