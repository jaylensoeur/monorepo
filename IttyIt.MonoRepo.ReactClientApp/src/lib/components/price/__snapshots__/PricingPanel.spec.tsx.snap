// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PricingPanel /> should render 1`] = `
<div
  className="makeStyles-pricePanel-1"
>
  <sup
    data-testid="price-description"
  >
     
    
  </sup>
  <div
    data-testid="currency-price"
  >
    <sup
      className="makeStyles-currencySymbol-3"
    >
      $
    </sup>
    <span
      className="makeStyles-price-7"
      data-testid="price"
    >
      250
    </span>
  </div>
  <div
    data-testid="approx-savings"
  >
    
  </div>
</div>
`;
